libcudacxx/test/libcudacxx/cuda/numeric/overflow.arithmetic/div_overflow.pass.cpp

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the libcu++ Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/numeric>
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+#include <cuda/std/utility>
+
+#include "test_macros.h"
+
+template <typename Result, typename Lhs, typename Rhs>
+__host__ __device__ constexpr void
+test_div_overflow(const Lhs lhs, const Rhs rhs, bool overflow, bool special_case = false, Result expected = {})
+{
+  // test overflow_result<Result> div_overflow(Lhs lhs, Rhs rhs) overload
+  //   skip_special_case is used to skip special cases that are not valid in a constexpr context, e.g. INT_MIN / -1
+  //   but the result fits in the Result type
+  {
+    const auto result = cuda::div_overflow<Result>(lhs, rhs);
+    if (!overflow)
+    {
+      if (special_case)
+      {
+        assert(result.value == expected);
+      }
+      else
+      {
+        assert(result.value == static_cast<Result>(static_cast<Result>(lhs) / static_cast<Result>(rhs)));
```

Comment:
nit: Are the  `static_cast<Result>`s on `lhs` and `rhs` necessary here? I see that our test cases are concrete at the moment e.g:

  ```cpp
 // 3. 0 / 1 -> should not overflow
  test_div_overflow<Result>(Lhs{0}, Rhs{1}, false);
```

but wouldn't a test case like 

```cpp
  test_div_overflow<Result>(Lhs{0}, Rhs{2^31 + 1}, false);
```
lead to undefined behavior due potential truncation ?